src/main/java/cashleh/CashLeh.java

```diff
@@ -25,14 +32,18 @@ public void run() {
         String inputString = input.getInputString();
         ui.printText("Hello " + inputString);
 
-        do {
+        Command command = null;
+        while (!(command instanceof Exit)) {
             inputString = input.getInputString();
-            if (inputString.equals("bye")) {
-                ui.printText("Bye. Hope to see you again soon!");
-            } else {
-                ui.printText("Sorry, I don't understand what you mean.");
+            try {
+                command = parser.parse(inputString);
+                command.execute(expenseStatement, incomeStatement);
```

Comment:
I think that we should change the implementation such that the statements are passed into the parser, as not all commands need both statements to function